spirv_common.hpp

```diff
@@ -296,6 +296,8 @@ struct SPIRType : IVariant
 		bool depth;
 		bool arrayed;
 		bool ms;
+		bool is_read = false;
```

Comment:
@HansKristian-ARM 

Agreed.

The challenge is that in MSL R/W access is declared as part of the texture type.

See  [SPIRV-Cross/reference/shaders-msl/desktop-only/frag/image-ms.desktop.frag](https://github.com/brenwill/SPIRV-Cross/blob/c1b8154f2ca1c82579cea53a646e37ef79bbea3d/reference/shaders-msl/desktop-only/frag/image-ms.desktop.frag).

Generation of this is currently buried in `image_type_glsl()` which is called from `type_to_glsl()`.

I'm not sure how to handle this without making it too disruptive everywhere.

Perhaps we could pass an optional `var_id` param to those two functions where they could look up decorations if it is non-zero?

Thoughts or suggestions?